File: src/_01_CreationalPattern/_01_04_Builder/java/after/DetailPlan.java
package _01_CreationalPattern._01_04_Builder.java.after;

public class DetailPlan {
  
  private int day;
  private String plan;

  public DetailPlan(int day, String plan) {
    this.day = day;
    this.plan = plan;
  }

  // day
  public int getDay() {
    return day;
  }
  public void setDay(int day) {
    this.day = day;
  }

  // plan
  public String getPlan() {
    return plan;
  }
  public void setPlan(String plan) {
    this.plan = plan;
  }

  @Override
  public String toString() {
    return "DetailPlan {"
      + " day=" + day
      + ", plan=\"" + plan + "\""
      + " }";
  }
}

File: src/_01_CreationalPattern/_01_04_Builder/DetailPlan.js
export default class DetailPlan {
  /**  @type { number } */
  #day

  /** @type { string } */
  #plan

  /**
   * @param { number } day 
   * @param { string } plan 
   */
  constructor(day, plan) {
    this.#day = day;
    this.#plan = plan;
  }

  /** @returns { number } */
  getDay() {
    return this.#day;
  }
  /** @param { number } day */
  setDay(day) {
    this.#day = day;
  }

  /** @return { string } */
  getPlan() {
    return this.#plan;
  }
  /** @param { string } plan */
  setPlan(plan) {
    this.#plan = plan;
  }

  /** @returns { string } */
  printInfo() {
    return `
      \t\tDetailPlan { day: ${this.#day}, plan: ${this.#plan} }
    `;
  }
}